### PR TITLE
change: add account number and unit tests for govcloud

### PR DESCRIPTION
**Pull request by yangaws**
*Issue #, if available:*

*Description of changes:*
Add account number for govcloud with unit tests

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

- [x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [ ] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#commit-message-guidlines)
- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have updated any necessary [documentation](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/sagemaker/amazon/amazon_estimator.py [hunk 1]
```diff
@@ -352,17 +355,20 @@ def registry(region_name, algorithm=None):
...
             'eu-west-1': '802834080501',
-            'us-east-2': '007439368137'
+            'us-east-2': '007439368137',
         }[region_name]
     else:
         raise ValueError("Algorithm class:{} doesn't have mapping to account_id with images".format(algorithm))
-    return "{}.dkr.ecr.{}.amazonaws.com".format(account_id, region_name)
+
+    domain_name = "c2s.ic.gov" if region_name == "us-iso-east-1" else "amazonaws.com"
+    return "{}.dkr.ecr.{}.{}".format(account_id, region_name, domain_name)
```
Comment (laurenyu) [inline src/sagemaker/amazon/amazon_estimator.py]: nit: use single quotes

File: src/sagemaker/fw_registry.py [hunk 2]
```diff
@@ -80,7 +84,8 @@ def registry(region_name, framework=None):
     """
     try:
         account_id = image_registry_map[region_name][framework]
-        return "{}.dkr.ecr.{}.amazonaws.com".format(account_id, region_name)
+        domain_name = "c2s.ic.gov" if region_name == "us-iso-east-1" else "amazonaws.com"
+        return "{}.dkr.ecr.{}.{}".format(account_id, region_name, domain_name)
```
Comment (laurenyu) [inline src/sagemaker/fw_registry.py]: should we maybe refactor this logic somewhere?

File: src/sagemaker/amazon/amazon_estimator.py (hunk 1 above)
Comment (yangaws) [reply src/sagemaker/amazon/amazon_estimator.py]: I usually apply the principle to use same quotes as nearby codes. Codes in this file use double quotes. Do you want to me to change all of them?

Comment (yangaws) [reply src/sagemaker/amazon/amazon_estimator.py]: Changed all of them.

File: src/sagemaker/fw_registry.py (hunk 2 above)
Comment (yangaws) [reply src/sagemaker/fw_registry.py]: Updated. Wrapped in one function in utils.py

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: SUCCEEDED
* [Build Logs](https://9dzx6wxcy9.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=87c8ee38-46d0-4094-bc80-4ad12cb1d22b%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Comment by jesterhazy** [comment]
### AWS CodeBuild CI Report

* Result: SUCCEEDED
* [Build Logs](https://9dzx6wxcy9.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=a5a99f32-047f-4835-8425-be18343bbbed%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

**Pull request merged**